golem-debugging-service/src/oplog/debug_oplog.rs

```diff
@@ -129,9 +129,8 @@ impl Oplog for DebugOplog {
         self.inner.length().await
     }
 
-    async fn upload_payload(&self, _data: &[u8]) -> Result<OplogPayload, String> {
-        // in a debugging session we don't need to upload anything
-        Err("Workers in debug cannot upload any new data to oplog".to_string())
+    async fn upload_payload(&self, data: &[u8]) -> Result<OplogPayload, String> {
```

Comment:
Going past the last entry in the oplog could cause the oplog rpc to fail as there was a race between this and the check_interrupt mechanism for trapping the worker.